### ISSUE TITLE
"Harvest Sources" root nav is marked as selected within /datasets (and vice-versa)

### ISSUE THREAD

**Issue by johnmartin** [GSA/ckanext-geodatagov#7]
OK this one is a complicated one. I'm going to write a quick fix within `ckanext-geodatagov` but we really probably should fix this in `ckan` or `ckanext-harvest`

Here's some background:

On `ckanext-geodatagov` we've added "Harvest Sources" as a main nav item. See: https://github.com/okfn/ckanext-geodatagov/blob/master/ckanext/geodatagov/templates/header.html#L86

Now when you visit http://geo.gov.ckan.org/ and visit the harvest source or dataset index pages you get:

![Screen Shot 2013-03-12 at 14 33 27](https://f.cloud.github.com/assets/67366/249028/5ff6f05e-8b22-11e2-93af-6bd593bc1047.png)

Not good.

After some investigation. It all comes down to this: https://github.com/okfn/ckan/blob/master/ckan/lib/helpers.py#L302

Basically because `_link_active` is returning true in both instances because within `ckanext-harvest` the routes `harvest_search` and `package_search` are the same controller + action.

@amercader @tobes I'll write the template fix quickly now, but what's the correct fix for this behaviour?
